### PR TITLE
Change expected exception to ParseException 

### PR DESCRIPTION
**Pull request by Zain-Abbas1**
Change expected exception to correct one

### Type of change

* [x] Bug fix
* [ ] New feature
* [ ] Breaking change

### Pull request checklist

* [ ] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [ ] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

File: src/test/java/uk/gov/ch/oauth/OAuth2StateHandlerTest.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ void oauth2DecodeState() {
     void oauth2DecodeHandlesExceptionAndLogs() {
         String encoded = "words.words.words.words.words";
         assertNull(stateHandler.oauth2DecodeState(encoded));
-        verify(mockLogger).error(matches("Could not decode OAuth state"), any(JOSEException.class));
+        verify(mockLogger).error(matches("Could not decode OAuth state"), any(ParseException.class));
```
Comment (CH-JonathanBlake) [inline src/test/java/uk/gov/ch/oauth/OAuth2StateHandlerTest.java:89]: Please check commit 
https://github.com/companieshouse/oauth-signin-java-library/pull/15/commits/ee2405970dcafc7b31845013de25ffc696b6944e

How will these changes cause merge conflicts going forwards?

Comment (CH-JonathanBlake) [reply src/test/java/uk/gov/ch/oauth/OAuth2StateHandlerTest.java:89]: I chose to make them completely generic becuase the test was that it handles exceptions not that it handles a specific type of exception.

Comment (Zain-Abbas1) [reply src/test/java/uk/gov/ch/oauth/OAuth2StateHandlerTest.java:89]: If Oauth is expecting Exception then you won't see the failure, so no need to change OAuth

Comment (Zain-Abbas1) [reply src/test/java/uk/gov/ch/oauth/OAuth2StateHandlerTest.java:89]: But it's good to verify for the specific exception so oauth might need to be updated

Comment (CH-JonathanBlake) [reply src/test/java/uk/gov/ch/oauth/OAuth2StateHandlerTest.java:89]: We're over specifing the exception. Testing its specifically a parse exception is a test of how the JWEObject is constructed.

**Pull request merged**